lib/engine/step/g_1860/buy_cert.rb

```diff
@@ -0,0 +1,268 @@
+# frozen_string_literal: true
+
+require_relative '../base'
+require_relative '../../action/par'
+
+module Engine
+  module Step
+    module G1860
+      class BuyCert < Base
+        attr_reader :companies, :in_auction
+
+        ALL_ACTIONS = %w[bid pass par].freeze
+        AUCTION_ACTIONS = %w[bid pass].freeze
+        PASS_ACTION = %w[pass].freeze
+        MIN_BID_RAISE = 5
+
+        def setup
+          @companies = @game.companies.select { |c| c.all_abilities.any? }.sort +
+                       @game.corporations.select { |c| c.layer == 1 }
+          @bids = {}
+          setup_auction
+        end
+
+        def available
+          @companies.select { |c| can_afford?(current_entity, c) }
+        end
+
+        def may_purchase?(_company)
+          true
+        end
+
+        def auctioning
+          :turn if @in_auction
```

Comment:
why do you make this :turn instead of just true or false?

Reply:
No particular reason, other than being more explicit about the reason for the check in the auction view.

Reply:
let's do 

def auctioning?
...
end

and then in all the code instead of referencing in_auction, just calling auctioning?

Reply:
In other games' auctions, it is either nil or a company.

Reply:
I'll still need to def auctioning, since the auctioning view calls @step.auctioning and it expects either nil or a company.

Reply:
ah i see what you're doing, ok let's keep it then